pkg/kubelet/cri/remote/remote_runtime.go

```diff
@@ -128,8 +128,8 @@ func (r *remoteRuntimeService) useV1API() bool {
 // being upgraded, then the container runtime must also support the initially
 // selected version or the redial is expected to fail, which requires a restart
 // of kubelet.
-func (r *remoteRuntimeService) determineAPIVersion(conn *grpc.ClientConn, endpoint string) error {
-	ctx, cancel := getContextWithTimeout(r.timeout)
+func (r *remoteRuntimeService) determineAPIVersion(ctx context.Context, conn *grpc.ClientConn, endpoint string) error {
+	ctx, cancel := context.WithTimeout(ctx, r.timeout)
```

Comment:
and adding an additional timeout here , from r.timeout

Reply:
same comment as above.  I don't think this is the cause

Reply:
yeah, I was just commenting the parts I don´t understand 😄